### ISSUE TITLE
Enhanced Picture Remixer

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#9]
From MM-NEATv2: https://github.com/schrum2/MM-NEATv2/issues/359

**Comment by schrum2** [comment]
I've been applying the neural style algorithm (https://github.com/anishathalye/neural-style) with CPPN art as the style file. At first, the results were always amorphous blobs, but when I reduced the --style-layer-weight-exp parameter to 0.2, it actually produced something that looked like the original picture, but had color influences from the CPPN art. Furthermore, it only requires 100 iterations of the style algorithm, which is great.

This results propose a new means of remixing art with CPPNs. The neural style algorithm depends on VGG19, which is part of DL4J. So, if the user supplies an image, and CPPNs are being evolved, then I can presumably translate the neural style algorithm into Java and run it on the user-supplied image with each evolved CPPN image as the style. The interactive process might be quite slow, but I still think this is a cool idea to try.

**Comment by schrum2** [comment]
Need to wait on response to https://github.com/deeplearning4j/deeplearning4j/issues/4291 to move forward

**Comment by schrum2** [comment]
I've computed all of the loss functions, but now I need to compute the gradient and perform the optimization. Unfortunately, I'm not sure DL4J is set up to handle the optimization of anything besides a neural network. I've asked for help on this in the DL4J Gitter channel, so we'll see what they say.

Otherwise, I may have to reimplement L-BGFS on my own ...

**Comment by schrum2** [comment]
I was asked to post my Gitter question as an Issue on DL4J, so I did:
https://github.com/deeplearning4j/deeplearning4j/issues/4298

**Comment by schrum2** [comment]
Totally abandoned DL4J approach, and have the approach that calls a Python program technically working, but slow. Ideas for speed up:

- Reduce number of style transfer iterations (not necessary for CPPN images anyway). This should probably be a slider anyway.
- Change Python code to create on TensorFlow session and reuse it instead of setting up each time.
- In Python, round off the Floats to Ints before sending the json image to stdout ... could save some I/O time

**Comment by schrum2** [comment]
Should add sliders to the interface to change the number of iterations and the style component weight ... and maybe other parameters.

Also, still need to look for ways to speed up the TensorFlow code for repeated use.

Lastly, and all Interactive Evolution Tasks can benefit from this, selected items that carry over to the next generation do not need to be re-processed. I can save a lot of time by simply moving the image over (I already generated it!).

**Comment by schrum2** [comment]
Made several improvements to speed up, like caching generated images and not wasting time on recalculations in Python code, but it is still a bit slow. Still, it works!
